# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/rustc/arm.patch
# Copyright (C) 2022 - 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- rustc-1.78.0-src/compiler/rustc_target/src/spec/targets/armv5te_unknown_linux_gnueabihf.rs.vanilla	2024-05-18 18:32:23.168970273 +0200
+++ rustc-1.78.0-src/compiler/rustc_target/src/spec/targets/armv5te_unknown_linux_gnueabihf.rs	2024-05-18 18:34:57.762962486 +0200
@@ -2,7 +2,7 @@
 
 pub fn target() -> Target {
     Target {
-        llvm_target: "armv5te-unknown-linux-gnueabi".into(),
+        llvm_target: "armv5te-unknown-linux-gnueabihf".into(),
         metadata: crate::spec::TargetMetadata {
             description: None,
             tier: None,
@@ -13,8 +13,8 @@
         data_layout: "e-m:e-p:32:32-Fi8-i64:64-v128:64:128-a:0:32-n32-S64".into(),
         arch: "arm".into(),
         options: TargetOptions {
-            abi: "eabi".into(),
-            features: "+soft-float,+strict-align".into(),
+            abi: "eabihf".into(),
+            features: "+vfp2,+strict-align".into(),
             // Atomic operations provided by compiler-builtins
             max_atomic_width: Some(32),
             mcount: "\u{1}__gnu_mcount_nc".into(),

--- rustc-1.78.0-src/compiler/rustc_target/src/spec/targets/armv6_unknown_linux_gnueabihf.rs.vanilla	2024-05-18 18:35:36.792960520 +0200
+++ rustc-1.78.0-src/compiler/rustc_target/src/spec/targets/armv6_unknown_linux_gnueabihf.rs	2024-05-18 18:35:45.747960069 +0200
@@ -2,7 +2,7 @@
 
 pub fn target() -> Target {
     Target {
-        llvm_target: "arm-unknown-linux-gnueabihf".into(),
+        llvm_target: "armv6-unknown-linux-gnueabihf".into(),
         metadata: crate::spec::TargetMetadata {
             description: None,
             tier: None,
--- rustc-1.78.0-src/compiler/rustc_target/src/spec/targets/armv6eb_unknown_linux_gnueabi.rs.vanilla	2024-05-18 18:36:22.238958230 +0200
+++ rustc-1.78.0-src/compiler/rustc_target/src/spec/targets/armv6eb_unknown_linux_gnueabi.rs	2024-05-18 18:36:49.105956877 +0200
@@ -3,7 +3,7 @@
 
 pub fn target() -> Target {
     Target {
-        llvm_target: "armeb-unknown-linux-gnueabi".into(),
+        llvm_target: "armv6eb-unknown-linux-gnueabi".into(),
         metadata: crate::spec::TargetMetadata {
             description: None,
             tier: None,
@@ -15,7 +15,7 @@
         arch: "arm".into(),
         options: TargetOptions {
             abi: "eabi".into(),
-            features: "+strict-align,+v8,+crc".into(),
+            features: "+strict-align,+v6,+soft-float,+crc".into(),
             endian: Endian::Big,
             max_atomic_width: Some(64),
             mcount: "\u{1}__gnu_mcount_nc".into(),
--- rustc-1.78.0-src/compiler/rustc_target/src/spec/targets/armv6eb_unknown_linux_gnueabihf.rs.vanilla	2024-05-18 18:37:02.021956227 +0200
+++ rustc-1.78.0-src/compiler/rustc_target/src/spec/targets/armv6eb_unknown_linux_gnueabihf.rs	2024-05-18 18:38:04.987953055 +0200
@@ -3,7 +3,7 @@
 
 pub fn target() -> Target {
     Target {
-        llvm_target: "armeb-unknown-linux-gnueabi".into(),
+        llvm_target: "armebv6-unknown-linux-gnueabihf".into(),
         metadata: crate::spec::TargetMetadata {
             description: None,
             tier: None,
@@ -14,8 +14,8 @@
         data_layout: "E-m:e-p:32:32-Fi8-i64:64-v128:64:128-a:0:32-n32-S64".into(),
         arch: "arm".into(),
         options: TargetOptions {
-            abi: "eabi".into(),
-            features: "+strict-align,+v8,+crc".into(),
+            abi: "eabihf".into(),
+            features: "+strict-align,+v6,+vfp2,+crc".into(),
             endian: Endian::Big,
             max_atomic_width: Some(64),
             mcount: "\u{1}__gnu_mcount_nc".into(),
